modules/http-codec.lua

```diff
@@ -88,7 +88,7 @@ exports.encoder = function ()
   local encodeHead, encodeRaw, encodeChunked
 
   function encodeHead(item)
-    if not item then return end
+    if not item or item == "" then return end
```

Comment:
I think the correct thing here is to `return item`.  If the user passes in `""` to the encoder it's basically a no-op and shouldn't end the stream like `nil` does.  By emitting an empty string we're essentially telling the socket to write nothing.